### PR TITLE
Policy management feature

### PR DESCRIPTION
**Pull request by angelmoo**
Changes for new a new vcert feature, “policy management”, This chances contains:
  >  Creating policy from a policy specification file on TPP and Venafi cloud.
  >  Modifying a policy using a policy specification file on TPP and Venafi cloud.
  >  Getting a policy specification details fromTPP and Venafi cloud.
  > Test suite for testing this new feature.

### CODE REVIEW

**Review by rvelaVenafi — APPROVED** [review]
Code has been carefully reviewed for the past week or two. Looks good. Minor issues were previously reported and fixed.

File: pkg/policy/policyUtils.go [hunk 1]
```diff
@@ -0,0 +1,1281 @@
...
+var TppKeyType = []string{"RSA", "ECDSA"}
+var TppRsaKeySize = []int{512, 1024, 2048, 3072, 4096}
+var CloudRsaKeySize = []int{1024, 2048, 4096}
+var TppEllipticCurves = []string{"P256", "P384", "P521"}
+
+func GetFileType(f string) string {
+	extension := filepath.Ext(f)
+
+	//As yaml extension could be yaml or yml then convert it to just .yaml
+	if extension == ".yml" {
```
Comment (marcos-albornoz) [inline pkg/policy/policyUtils.go:21]: if the extension is in uppercase, then maybe will be great to support it

File: pkg/venafi/cloud/connector.go [hunk 2]
```diff
@@ -72,6 +78,213 @@ type Connector struct {
 	client  *http.Client
 }
 
+func (c *Connector) GetPolicySpecification(name string) (*policy.PolicySpecification, error) {
+
+	appName := policy.GetApplicationName(name)
+	if appName != "" {
+		c.zone.appName = appName
+	} else {
+		return nil, fmt.Errorf("application name is not valid, please provice a valid zone name in the format: appName\\CitName")
```
Comment (marcos-albornoz) [inline pkg/venafi/cloud/connector.go:87]: misspelling in "provice" word

File: pkg/venafi/cloud/connector.go [hunk 3]
```diff
@@ -72,6 +78,213 @@ type Connector struct {
...
+	}
+	citName := policy.GetCitName(name)
+	if citName != "" {
+		c.zone.templateAlias = citName
+	} else {
+		return nil, fmt.Errorf("cit name is not valid, please provice a valid zone name in the format: appName\\CitName")
```
Comment (marcos-albornoz) [inline pkg/venafi/cloud/connector.go:93]: misspelling in "provice" word

File: cmd/vcert/commands.go [hunk 4]
```diff
@@ -611,6 +635,159 @@ func doCommandRevoke1(c *cli.Context) error {
...
+	}
+
+	_, err = connector.SetPolicy(policyName, &policySpecification)
+
+	defer file.Close()
+
+	return err
+}
+
+func doCommandGetPolicy1(c *cli.Context) error {
```
[the remark this answers is not in the document]
Comment (angelmoo) [reply cmd/vcert/commands.go:710]: @tr1ck3r all the commands have the 1 at the end of the name, should we keep this or should I remove at least for this function the 1 at the name of the function.

File: cmd/vcert/validators.go [hunk 5]
```diff
@@ -643,3 +643,51 @@ func validateOverWritingEnviromentVariables() {
 	}
 
 }
+
+func validateGetPolicyFlags1(commandName string) error {
```
[the remark this answers is not in the document]
Comment (angelmoo) [reply cmd/vcert/validators.go:647]: @tr1ck3r all the validation commands have the 1 at the end of the name, should we keep this or should I remove at least for this function the 1 at the name of the function.

File: cmd/vcert/commands.go [hunk 6]
```diff
@@ -611,6 +635,159 @@ func doCommandRevoke1(c *cli.Context) error {
 	return nil
 }
 
+func doCommandCreatePolicy1(c *cli.Context) error {
```
[the remark this answers is not in the document]
Comment (angelmoo) [reply cmd/vcert/commands.go:638]: @tr1ck3r all the commands have the 1 at the end of the name, should we keep this or should I remove at least for this function the 1 at the name of the function.

File: cmd/vcert/validators.go [hunk 7]
```diff
@@ -643,3 +643,51 @@ func validateOverWritingEnviromentVariables() {
...
+		}
+
+		if flags.policyName == "" {
+			return fmt.Errorf("zone is required")
+		}
+	}
+	return nil
+}
+
+func validateSetPolicyFlags1(commandName string) error {
```
[the remark this answers is not in the document]
Comment (angelmoo) [reply cmd/vcert/validators.go:666]: @tr1ck3r all the commands have the 1 at the end of the name, should we keep this or should I remove at least for this function the 1 at the name of the function.

File: cmd/vcert/commands.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (angelmoo) [reply cmd/vcert/commands.go]: fixed removed 1 at the end

File: pkg/policy/constants.go [hunk 8]
```diff
@@ -0,0 +1,44 @@
+package policy
+
+const (
+	JsonExtention        = ".json"
```
[the remark this answers is not in the document]
Comment (angelmoo) [reply pkg/policy/constants.go]: fixed

File: pkg/policy/constants.go [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+	YamlExtention        = ".yaml"
```
[the remark this answers is not in the document]
Comment (angelmoo) [reply pkg/policy/constants.go]: fixed

File: pkg/policy/policyUtils.go (hunk 1 above)
Comment (angelmoo) [reply pkg/policy/policyUtils.go:22]: fixed

File: pkg/venafi/cloud/connector.go (hunk 2 above)
Comment (angelmoo) [reply pkg/venafi/cloud/connector.go]: fixed on latest commit

File: pkg/venafi/cloud/connector.go (hunk 3 above)
Comment (angelmoo) [reply pkg/venafi/cloud/connector.go]: fixed on latest commit

File: cmd/vcert/validators.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (angelmoo) [reply cmd/vcert/validators.go]: fixed removed 1 at the end

**Review by marcos-albornoz — APPROVED** [review]

**Pull request merged**